File: C++/Medium/CompileError/closest-dessert-cost_7.cpp
class Solution {
public:
    int closestCost(vector<int>& baseCosts, vector<int>& toppingCosts, int target) {
        
        int ans = 0;
        for (int i = 0; i < baseCosts.size(); i++) {
            int base = baseCosts[i];
            ans = min(ans, base, target);
            for (int j = 0; j < toppingCosts.size(); j++) {
                int topping = toppingCosts[j];
                ans = min(ans, base + topping, target);
                for (int k = j + 1; k < toppingCosts.size(); k++) {
                    int topping2 = toppingCosts[k];
                    ans = min(ans, base + topping + topping2, target);
                }
            }
        }
        return ans;
    }
    
    int min(int a, int b, int target) {
        if (abs(a - target) < abs(b - target)) {
            return a;
        } else if (abs(a - target) > abs(b - target)) {
            return b;
        } else {
            return min(a, b);
        }
    }
};

File: Java/Medium/WrongAnswer/closest-dessert-cost_7.java
class Solution {
    public int closestCost(int[] baseCosts, int[] toppingCosts, int target) {
        
        int min = Integer.MAX_VALUE;
        int result = 0;
        for (int baseCost : baseCosts) {
            int sum = baseCost;
            int diff = Math.abs(target - sum);
            if (diff < min) {
                min = diff;
                result = sum;
            }
            for (int toppingCost : toppingCosts) {
                sum += toppingCost;
                diff = Math.abs(target - sum);
                if (diff < min) {
                    min = diff;
                    result = sum;
                }
                for (int toppingCost2 : toppingCosts) {
                    sum += toppingCost2;
                    diff = Math.abs(target - sum);
                    if (diff < min) {
                        min = diff;
                        result = sum;
                    }
                    sum -= toppingCost2;
                }
                sum -= toppingCost;
            }
        }
        return result;
    }
}